src/lib/raft/raft.c

```diff
@@ -494,6 +494,7 @@ raft_process_msg(struct raft *raft, const struct raft_msg *req, uint32_t source)
 			 * it manually.
 			 */
 			raft->leader = 0;
+			raft_schedule_broadcast(raft);
```

Comment:
But broadcast doesn't include `raft->leader`.
It only sends `term, vote and state`.
And we do not broadcast "who thinks what about leader liveliness".
Or am I missing something?

Reply:
This is for watcher events ("box.election"), not raft messages. Watcher built-in events will need updates on who is the leader. And they are sent from a trigger which is fired by `raft_schedule_broadcast()`.

Reply:
Oh, I see. Thanks for the explanation!
LGTM then.

Reply:
Maybe add a unit test then?